pkg/allocateip/allocateip.go

```diff
@@ -399,7 +583,14 @@ func removeHostTunnelAddr(ctx context.Context, c client.Interface, nodename stri
 
 // determineEnabledPools returns all enabled pools. If vxlan is true, then it will only return VXLAN pools. Otherwise
 // it will only return IPIP enabled pools.
-func determineEnabledPoolCIDRs(node api.Node, ipPoolList api.IPPoolList, vxlan bool) []net.IPNet {
+func determineEnabledPoolCIDRs(node api.Node, ipPoolList api.IPPoolList, attrType string) []net.IPNet {
+	// For wireguard, return no valid pools if the wireguard public key has not been set. Only once wireguard has been
+	// enabled *and* the wireguard device has been initialized do we require an IP address to be configured.
```

Comment:
Don't we need the IP in order to init the wg interface?

I think we'll end up with:
- Felix creates interface, writes back public key
- This code spots it, assigns IP
- Felix spots the IP, and writes it to the wg interface.

Is that right? It will work, it's just different from how we do VXLAN IIUC, since for VXLAN we don't create the interface until the IP has been allocated, since the overlay won't work properly until the IP is programmed on it.

Reply:
That is correct.  We can configure wg without an IP and then update it after.  I think this is OK because it's lightweight - in that felix doesn't do a full restart because the interface address was updated.